src/pq_flash_index.cpp

```diff
@@ -633,38 +630,38 @@ void PQFlashIndex<T, LabelT>::parse_label_file(const std::string &label_file, si
     get_label_file_metadata(label_file, num_pts_in_label_file, num_total_labels);
 
     _pts_to_label_offsets = new uint32_t[num_pts_in_label_file];
-    _pts_to_labels = new uint32_t[num_pts_in_label_file + num_total_labels];
-    uint32_t counter = 0;
+    _pts_to_label_counts = new uint32_t[num_pts_in_label_file];
+    _pts_to_labels = new LabelT[num_total_labels];
+    uint32_t labels_seen_so_far = 0;
 
     while (std::getline(infile, line))
     {
         std::istringstream iss(line);
         std::vector<uint32_t> lbls(0);
 
-        _pts_to_label_offsets[line_cnt] = counter;
-        uint32_t &num_lbls_in_cur_pt = _pts_to_labels[counter];
+        _pts_to_label_offsets[line_cnt] = labels_seen_so_far;
+        uint32_t &num_lbls_in_cur_pt = _pts_to_label_counts[line_cnt];
         num_lbls_in_cur_pt = 0;
-        counter++;
         getline(iss, token, '\t');
         std::istringstream new_iss(token);
         while (getline(new_iss, token, ','))
         {
             token.erase(std::remove(token.begin(), token.end(), '\n'), token.end());
             token.erase(std::remove(token.begin(), token.end(), '\r'), token.end());
             LabelT token_as_num = (LabelT)std::stoul(token);
-            if (_labels.find(token_as_num) == _labels.end())
-            {
-                _filter_list.emplace_back(token_as_num);
-            }
-            int32_t filter_num = get_filter_number(token_as_num);
-            if (filter_num == -1)
-            {
-                diskann::cout << "Error!! " << std::endl;
-                exit(-1);
-            }
-            _pts_to_labels[counter++] = filter_num;
+            /*            if (_labels.find(token_as_num) == _labels.end())
+                        {
+                            _filter_list.emplace_back(token_as_num);
+                        }
+                        int32_t filter_num = get_filter_number(token_as_num);
+                        if (filter_num == -1)
+                        {
+                            diskann::cout << "Error!! " << std::endl;
+                            exit(-1);
+                        } */
+            _pts_to_labels[labels_seen_so_far++] = (uint32_t)token_as_num;
```

Comment:
not sure if they have been added. All use default value i believe.

Reply:
we need to change "uint32_t" to "LabelT" to be "  _pts_to_labels[labels_seen_so_far++] = (LabelT)token_as_num;" because _pts_to_labels is "LabelT" array